### PR TITLE
chore: supporting auto versioning of the generated SDK

### PR DESCRIPTION
**Pull request by Hamzanael**
* [x] Code is up-to-date with the `main` branch.
* [x] You've successfully built and run the tests locally.
* [ ] There are new or updated unit tests validating the change.

### :pencil: Description

chore: supporting auto versioning of the generated SDK

### CODE REVIEW

File: .github/workflows/main-sdk-action.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      name:
+        description: 'generated sdk name'
+        required: true
+      fileAsBase64:
+        description: 'generated sdk file as base64 encoded'
+        required: true
+jobs:
+  prepare_sdk_version_if_not_provided:
+    if: ${{ github.event.inputs.version == '' }}
+    runs-on: macos-latest
```
Comment (worldtiki) [inline .github/workflows/main-sdk-action.yml:18]: seems unnecessary to use macos for a curl

File: .github/workflows/main-sdk-action.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    needs: [ generate_sdk_with_auto_version,generate_sdk_with_provided_version ]
+    uses: ./.github/workflows/github-release-actions.yml
+    secrets: inherit
+
+  update-bom:
+    needs: deploy_sdk
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update BOM
+        uses: fjogeleit/http-request-action@v1
```
Comment (worldtiki) [inline .github/workflows/main-sdk-action.yml:62]: not sure if we want to rely on a 3rd party to pass secrets around

File: .github/workflows/generate-sdk-action.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: Build SDK
```
Comment (worldtiki) [inline .github/workflows/generate-sdk-action.yml:1]: isnt the idiomatic way of doing this to use the maven deploy plugin?

**Review by anssari1 — APPROVED** [review]
run on ubuntu-latest

**Pull request merged**